src/Branching/perform-migration-from-svn-to-git.md

```diff
@@ -0,0 +1,183 @@
+![Subversion users](_img/perform-migration-from-svn-to-git/svn-log.png)
+
+*Git users*
+![git users](_img/perform-migration-from-svn-to-git/git-log.png)
+
+To extract a list of all SVN users, from the root of your local Subversion checkout, run this command:
+
+```
+svn log -q | awk -F '|' '/^r/ {sub("^ ", "", $2); sub(" $", "", $2); print $2" = "$2" <"$2">"}' | sort -u > authors-transform.txt
+```
+This command will retrieve all the log messages, extract the usernames, eliminate any duplicate usernames, sort the usernames and place them into a “authors-transform.txt” file. You can then edit each line in the file to create a mapping of SVN users to a well-formatted Git user. For example, you can map `almrangers = almrangers <almrangers> ` to `almrangers =  almrangers <almrangers@example.com>` .
+
+### Clone the Subversion repository using git-svn
+
+The following command will do the standard git-svn transformation using the authors-transform.txt file created in the previous step. It will place the Git repository in the "c:\mytempdir" folder in your local machine.
+```
+git svn clone ["SVN repo URL"] --prefix=svn/ --no-metadata --authors-file "authors-transform.txt" --stdlayout c:\mytempdir
+```
+
+> [!NOTE]
+>
+> The –prefix=svn/ is necessary because otherwise the tools can’t tell apart SVN revisions from imported ones. Setting a prefix (with a trailing slash) is strongly encouraged in any case, as your SVN-tracking refs will then be located at "refs/remotes/$prefix/", which is compatible with Git’s own remote-tracking ref layout (refs/remotes/$remote/).
+>
+> Setting a prefix is also useful if you wish to track multiple projects that share a common repository. By default, the prefix is set to origin/.
+
+If you are using the standard trunk, branches, tags layout you’ll just put –stdlayout. However if you have something different you may have to pass the –trunk, –branches, and –tags in to identify what is what. For example if your repository structure was trunk/companydir and you branched that instead of trunk, you would probably want ‘–trunk=trunk/companydir –branches=branches‘.
+
+```
+git svn clone ["SVN repo URL"] --prefix=svn/ --no-metadata --trunk=/trunk --branches=/branches --tags=/tags  --authors-file "authors-transform.txt" c:\mytempdir
+```
+
+> [!NOTE]
+>
+> This command can take a few minutes to several hours depending on the size of the SVN repository. Upon completion, you will have a Git checkout of your repository.
+
+### Convert version control-specific configurations
+
+If your svn repo was using svn:ignore properties, you can  convert this to a .gitignore file using:
+```
+cd c:\mytempdir
+git svn show-ignore > .gitignore
+git add .gitignore
+git commit -m 'Convert svn:ignore properties to .gitignore.'
+```
+> [!TIP]
+>
+> Read more about .gitignore [Ignore file changes with Git](
+https://docs.microsoft.com/en-us/vsts/git/tutorial/ignore-files?tabs=visual-studio)
+
+### Push repository to a bare git repository
```

Comment:
This section seems awkward, and I don't understand the value.  What's the value of creating a whole second repository?  Once we have a repository that was created by git-svn, why not just update the tags and change `trunk` -> `master` and push that to VSTS directly?

Otherwise, we should explain what the value of this is.

Reply:
@ethomson The intent is to have a bare Git repo so that user workspace information is not pushed up to VSTS repo. You are correct we can skip that step if this is not an issue